src/main/java/com/icthh/xm/ms/entity/repository/XmEntityRepository.java

```diff
@@ -30,130 +30,162 @@
     }
 
+    @Override
     public void deleteInBatch(Iterable<XmEntity> entities) {
         springXmEntityRepository.deleteInBatch(entities);
     }
 
+    @Override
     public void deleteAllInBatch() {
         springXmEntityRepository.deleteAllInBatch();
     }
 
+    @Override
     public XmEntity getOne(Long aLong) {
         return springXmEntityRepository.getOne(aLong);
     }
 
+    @Override
     public <S extends XmEntity> List<S> findAll(Example<S> example) {
         return springXmEntityRepository.findAll(example);
     }
 
+    @Override
     public <S extends XmEntity> List<S> findAll(Example<S> example, Sort sort) {
         return springXmEntityRepository.findAll(example, sort);
     }
 
+    @Override
     public Page<XmEntity> findAll(Pageable pageable) {
         return springXmEntityRepository.findAll(pageable);
     }
 
-    public XmEntity findOne(Long aLong) {
-        return springXmEntityRepository.findOne(aLong);
+    @Override
+    public Optional<XmEntity> findById(Long aLong) {
+        return springXmEntityRepository.findById(aLong);
     }
 
-    public boolean exists(Long aLong) {
-        return springXmEntityRepository.exists(aLong);
+    @Override
+    public boolean existsById(Long aLong) {
+        return springXmEntityRepository.existsById(aLong);
     }
 
+    @Override
     public long count() {
         return springXmEntityRepository.count();
     }
 
-    public void delete(Long aLong) {
-        springXmEntityRepository.delete(aLong);
+    @Override
+    public void deleteById(Long aLong) {
+        springXmEntityRepository.deleteById(aLong);
     }
 
+    @Override
     public void delete(XmEntity entity) {
         springXmEntityRepository.delete(entity);
     }
 
-    public void delete(Iterable<? extends XmEntity> entities) {
-        springXmEntityRepository.delete(entities);
+    @Override
+    public void deleteAll(Iterable<? extends XmEntity> entities) {
+        springXmEntityRepository.deleteAll(entities);
     }
 
+    @Override
     public void deleteAll() {
         springXmEntityRepository.deleteAll();
     }
 
-    public <S extends XmEntity> S findOne(Example<S> example) {
+    @Override
+    public <S extends XmEntity> Optional<S> findOne(Example<S> example) {
         return springXmEntityRepository.findOne(example);
     }
 
+    @Override
     public <S extends XmEntity> Page<S> findAll(Example<S> example, Pageable pageable) {
         return springXmEntityRepository.findAll(example, pageable);
     }
 
+    @Override
     public <S extends XmEntity> long count(Example<S> example) {
         return springXmEntityRepository.count(example);
     }
 
+    @Override
     public <S extends XmEntity> boolean exists(Example<S> example) {
         return springXmEntityRepository.exists(example);
     }
 
-    public XmEntity findOne(Specification<XmEntity> spec) {
+    @Override
+    public Optional<XmEntity> findOne(Specification<XmEntity> spec) {
```

Comment:
It's can broke leps!
Please resolve optional inside repository

Reply:
This is a much worse situation:

if we change to:
```
 public XmEntity findOne(Specification<XmEntity> spec) {
        return springXmEntityRepository.findOne(spec).orElse(null);
    }```

then compilation error raises:

```
'findOne(Specification<XmEntity>)' in 'com.icthh.xm.ms.entity.repository.XmEntityRepository' clashes with 'findOne(Specification<T>)' in 'org.springframework.data.jpa.repository.JpaSpecificationExecutor'; attempting to use incompatible return type
```

Reply:
remove override
it's proxy interface don't need to implement interface of spring crud repository

Reply:
done